### PR TITLE
Bigint

### PR DESCRIPTION
**Pull request by jialunzhang-psu**

### CODE REVIEW

File: bigint/bigint.mbti [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+// Values
+fn from_decimal_string(String) -> BigInt
+
+fn from_int(Int) -> BigInt
+
+fn from_int64(Int64) -> BigInt
+
+fn make_zero() -> BigInt
+
```
Comment (bobzhang) [inline bigint/bigint.mbti:11]: TODO: we will add literal syntax support, e.g, `0n`

File: bigint/bigint.mbti [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  op_equal(Self, Self) -> Bool
+  op_mod(Self, Self) -> Self
+  op_mul(Self, Self) -> Self
+  op_neg(Self) -> Self
+  op_sub(Self, Self) -> Self
+  to_decimal_string(Self) -> String
+}
+
+type Sign
+impl Sign {
```
Comment (bobzhang) [inline bigint/bigint.mbti:32]: Should this type be publicly exposed?

File: bigint/bigint.mbti [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+fn from_int64(Int64) -> BigInt
+
+fn make_zero() -> BigInt
+
+// Types and methods
+type BigInt
+impl BigInt {
+  compare(Self, Self) -> Int
+  debug_write(Self, Buffer) -> Unit
+  deep_clone(Self) -> Self
```
Comment (bobzhang) [inline bigint/bigint.mbti:17]: when `deep_clone` is needed given BigInt is pure?

**Review by bobzhang — COMMENTED** [review]
looks good to me in general.
It would be interesting to see the performance compared with implementation using externref bindings through V8

File: bigint/bigint.mbti [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  lsl(Self, Int) -> Self
+  lsr(Self, Int) -> Self
+  op_add(Self, Self) -> Self
+  op_div(Self, Self) -> Self
+  op_equal(Self, Self) -> Bool
+  op_mod(Self, Self) -> Self
+  op_mul(Self, Self) -> Self
+  op_neg(Self) -> Self
+  op_sub(Self, Self) -> Self
+  to_decimal_string(Self) -> String
```
Comment (bobzhang) [inline bigint/bigint.mbti:28]: should be called `to_string` directly?

File: bigint/bigint.mbti (hunk 1 above)
Comment (jialunzhang-psu) [reply bigint/bigint.mbti:11]: annotated

File: bigint/bigint.mbti (hunk 2 above)
Comment (jialunzhang-psu) [reply bigint/bigint.mbti]: fixed

File: bigint/bigint.mbti (hunk 3 above)
Comment (jialunzhang-psu) [reply bigint/bigint.mbti]: fixed

File: bigint/bigint.mbti (hunk 4 above)
Comment (jialunzhang-psu) [reply bigint/bigint.mbti:27]: Later we might have to_hex_string, etc. But given that decimal is the most common case, we can name it just `to_string`.

**Pull request merged**